core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -281,7 +281,7 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
           workflowId =>
             uploadMetadataToGCS(workflowRec.submissionId.toString, workflowId, petUser).recover{
               case e =>
-                logger.error(s"Failed to upload metadata for ${workflowRec.submissionId}/${workflowId}", e)
+                logger.error(s"Failed to upload metadata for ${workflowRec.submissionId}/${workflowId} due to ${e}")
```

Comment:
no, the old way was the right way

Reply:
I think the error information you are looking for is immediately following in the logs

Reply:
it should. But I don't see the cause anywhere in kibana..So I'm just changing it to log it as plain text.

Reply:
your problem is on line 298 below, you are passing in `None` for the execId which triggers a lookup to cromwell based on labels but it all happens before the labels are set. Look at how the original `ShardedHttpExecutionServiceCluster.callLevelMetadata` function is called as opposed to the new one `callLevelMetadataForCostCalculation`. The code looks in the db for the execId first, you need to do that.